packages/authgear-react-native/android/src/main/java/com/authgear/reactnative/AuthgearReactNativeModule.java

```diff
@@ -919,6 +931,88 @@ public void signAnonymousToken(String kid, String data, Promise promise) {
         }
     }
 
+    @ReactMethod
+    public void createDPoPPrivateKey(ReadableMap options, Promise promise) {
+        String kid = options.getString("kid");
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            promise.resolve(Arguments.createMap());
+            return;
+        }
+        String alias = this.formatDPoPKeyAlias(kid);
+        try {
+            this.generateKey(alias);
+            promise.resolve("");
+        } catch (Exception e) {
+            promise.reject(e);
+        }
+    }
+
+    @ReactMethod
+    public void signWithDPoPPrivateKey(ReadableMap options, Promise promise) {
+        String kid = options.getString("kid");
+        ReadableMap payload = options.getMap("payload");
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            promise.resolve("");
+            return;
+        }
+        String alias = this.formatDPoPKeyAlias(kid);
+        try {
+            KeyPair keyPair = this.getPrivateKey(alias);
+            WritableMap jwk = this.getJWK(keyPair, kid);
+            final WritableMap header = this.makeDPoPJWTHeader(jwk);
+            Signature signature = this.makeSignature(keyPair.getPrivate());
+            String jwt = this.signJWT(signature, header, payload);
+            promise.resolve(jwt);
+        } catch (Exception e) {
+            promise.reject(e);
+        }
+    }
+
+    @ReactMethod
+    public void checkDPoPPrivateKey(ReadableMap options, Promise promise) {
+        String kid = options.getString("kid");
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            promise.resolve(true);
+            return;
+        }
+        String alias = this.formatDPoPKeyAlias(kid);
+        try {
+            this.getPrivateKey(alias);
+            promise.resolve(true);
```

Comment:
catch KeyNotFoundException and resolve false, otherwise, reject with any other exception

Reply:
@louischan-oursky Not only KeyNotFoundException. Any other exception also indicates the key cannot be used, so it is still false.

Reply:
For example, what kind of exception?

Reply:
We don't know all exception that could be thrown. Whenever there is error, it indicates signing dpop proof is impossible. And as the purpose of this function is to check if it is possible to sign a dpop proof, any error means false.